007_team_5_agent_design/007_team_5_agent_design.tex

```diff
@@ -1 +1,181 @@
+
+\subsection*{Strategy}\label{sec:team5-messaging-strategy}
+The agent's approach to messaging is to adopt a ``passive observer'' role focussed on gathering information about other agents to organise itself with these agents in the tower. With more information at its disposal, the agent can make more informed decisions to balance its own individual utility with the collective utility of the agents in the tower.
+
+\ToDo{choose whether to say 12 or 8 ticks to send all messages}
+The agent sends three types of messages targeting each of the agents two floors above and below its own floor; it therefore takes 12 ticks to send all of its messages. These messages ask for the HP, intended food intake, and actual food intake of other agents. The decision of which message type to send to which floor is controlled by the Messaging Counter described in Table \ref{tab:team5-personality}. This counter is incremented each tick, and reset either every 25 ticks or at the end of each day, whichever comes first. This ensures that the agent is requesting up-to-date information regularly for use in its decision making.
+
+The order of sent messages is:
+\begin{enumerate}
+    \item Ask for an agent's HP
+    \item Ask for an agent's previous food intake
+    % \item Ask for an agent's intended food intake
+\end{enumerate}
+The Team 5 agent first targets the agent immediately below it, then the agent immediately above, then two floors below, and finally the agent two floors above.
+
+\section{Social Memory}\label{sec:team5-memory}
+The agent stores the information gained from communication in its `memory': the information stored by this data structure is shown in Table \ref{tab:team5-memory}.
+
+\begin{table}
+    \centering
+    \begin{tabular}%
+        {| >{\raggedleft\arraybackslash}p{0.25\linewidth} | %
+        >{\raggedright\arraybackslash}p{0.65\linewidth} | %
+        }
+        \hline
+        Agent ID & An agent's unique ID\\
+        \hline
+        Agent HP & The agent's HP level\\
+        \hline
+        Food Taken & The amount of food the agent last took\\
+        % \hline
+        % Intended Food Intake & The amount of food the agent intends to take\\
+        \hline
+        Favour & The Team 5 agent's opinion of this agent\\
+        \hline
+        Days Since Last Seen & The number of days since the last message from this agent\\
+        \hline
+    \end{tabular}
+    \caption{Team 5 Agent Memory}
+    \label{tab:team5-memory}
+\end{table}
+
+This memory allows the agent to construct a social network of agents in the tower and evaluate others based on these parameters. This consequently leads to the idea of favour, a metric the Team 5 agent uses to judge other agents. If more than five days have passed since the agent's last interaction with someone, then the memory of that agent is reset. This prevents the agent from using out-of-date information in its calculation and also allows for the removal of dead agents from its memory.
```

Comment:
```suggestion
This memory allows the agent to construct a social network of agents in the tower and evaluate others based on these parameters. This consequently leads to the idea of favour, a metric the Team 5 agent uses to judge other agents. If more than five days have passed since the agent's last interaction with an agent in its social memory, then the social memory regarding that agent is reset. This prevents the agent from using out-of-date information in its calculation and also allows for the removal of dead agents from the social memory.
```

Reply:
```suggestion
This memory allows the agent to construct a social network of agents in the tower and evaluate others based on these parameters. This consequently leads to the idea of favour, a metric the Team 5 agent uses to judge other agents. If more than five days have passed since the agent's last interaction with another agent, then the memory of that agent is reset. This prevents the agent from using out-of-date information in its calculation and also allows for the removal of dead agents from its memory.
```